### PR TITLE
Refactoring flexible server create

### PR DESCRIPTION
**Pull request by arde0708**
This PR covers : 
1. Firewall naming updation based on IP's being added.
2. Refactoring flexible_server_create to avoid code path trigerred by an Exception. - PG and MySQL
3. handle edge cases in create experience.
4. Add more validations
5. Update connection port for Mysql connection string for PhP.
6. Fixes from bug bash.
7. update username in local context for MySQL connection string

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py [hunk 1]
```diff
@@ -28,76 +28,87 @@ def _flexible_server_create(cmd, client, resource_group_name=None, server_name=N
...
+
+            # Create mysql server
+            # Note : passing public_access has no effect as the accepted values are 'Enabled' and 'Disabled'. So the value ends up being ignored.
+            server_result = _create_server(db_context, cmd, resource_group_name, server_name, location, backup_retention,
+                                           sku_name, tier, storage_mb, administrator_login, administrator_login_password,
+                                           version, tags, delegated_subnet_arguments, assign_identity, public_access,
+                                           high_availability, zone)
+
+            # Adding firewall rule
+            if public_access is not None and public_access != 'none':
```
Comment (mjain2) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py]: .tolower() needed? If customer passes in something like "NONE", this won't be caught

Comment (mjain2) [reply src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py]: same in the elif

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_postgres.py [hunk 2]
```diff
@@ -31,68 +31,79 @@ def _flexible_server_create(cmd, client,
...
+                delegated_subnet_arguments = None
+            # Create postgresql
+            # Note : passing public_access has no effect as the accepted values are 'Enabled' and 'Disabled'. So the value ends up being ignored.
+            server_result = _create_server(db_context, cmd, resource_group_name, server_name, location, backup_retention,
+                                           sku_name, tier, storage_mb, administrator_login, administrator_login_password,
+                                           version, tags, subnet_id, assign_identity,delegated_subnet_arguments,
+                                           high_availability, zone)
+
+            # Adding firewall rule
+            if public_access is not None and public_access != 'none':
```
Comment (mjain2) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_postgres.py]: same as earlier comment - potential tolower() needed

File: src/azure-cli/azure/cli/command_modules/rdbms/validators.py [hunk 3]
```diff
@@ -200,8 +200,8 @@ def ip_address_validator(ns):
 def public_access_validator(ns):
     if ns.public_access:
         val = ns.public_access
-        if not(val == 'on' or (len(val.split('-')) == 1 and _validate_ip(val)) or (len(val.split('-')) == 2 and _validate_ip(val))):
-            raise CLIError('Incorrect usage: --public-access. Acceptable values are \'on\' or \'<startIP>\' or \'<startIP>-<destinationIP>\' where startIP and destinationIP ranges from 0.0.0.0 to 255.255.255.255')
+        if not (val == 'all' or val == 'none' or (len(val.split('-')) == 1 and _validate_ip(val)) or (len(val.split('-')) == 2 and _validate_ip(val))):
```
Comment (mjain2) [inline src/azure-cli/azure/cli/command_modules/rdbms/validators.py:203]: again, does case matter?

**Review by mjain2 — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py (hunk 1 above)
Comment (arde0708) [reply src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py]: Addressed

**Pull request merged**